lib/parser/file_ref.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Thrift.Parser.FileRef do
+  @type thrift_include :: String.t
+  @type t :: %__MODULE__{path: String.t, include_name: String.t, contents: String.t}
+  defstruct path: nil, include_name: nil, contents: nil
+
+  def new(path) do
+    thrift_file = File.read!(path) <> "\n__file__ \"#{path}\""
```

Comment:
It gets used when we're building the names of things. The thrift module is the same as the file name, so we need the filename somewhere. However, leex and yecc deal with byte streams, and there's no way to get the file. 
This is a hack around that.